cmd/monitor/main.go

```diff
@@ -0,0 +1,138 @@
+	once := flag.Bool("once", true, "whether to run the monitor on a repeated interval or once")
+	flag.Parse()
+
+	if configFilePath == nil {
+		log.Fatalf("empty configuration file path")
+	}
+
+	readConfig, err := os.ReadFile(*configFilePath)
+	if err != nil {
+		log.Fatalf("error reading from identity monitor configuration file: %v", err)
+	}
+
+	configString := string(readConfig)
+	var config IdentityMonitorConfiguration
+	if err := yaml.Unmarshal([]byte(configString), &config); err != nil {
+		log.Fatalf("error parsing identities: %v", err)
+	}
+
+	rekorClient, err := client.GetRekorClient(config.ServerURL, client.WithUserAgent(strings.TrimSpace(fmt.Sprintf("rekor-monitor/%s (%s; %s)", version.GetVersionInfo().GitVersion, runtime.GOOS, runtime.GOARCH))))
+	if err != nil {
+		log.Fatalf("getting Rekor client: %v", err)
+	}
+
+	if config.ServerURL == "" {
+		config.ServerURL = publicRekorServerURL
+	}
+	if config.LogInfoFile == "" {
+		config.LogInfoFile = logInfoFileName
+	}
+	if config.OutputIdentitiesFile == "" {
+		config.OutputIdentitiesFile = outputIdentitiesFileName
+	}
+	if config.Interval == nil {
+		defaultInterval := time.Hour
+		config.Interval = &defaultInterval
+	}
+
+	ticker := time.NewTicker(*config.Interval)
+	defer ticker.Stop()
+
+	// To get an immediate first tick
+	for ; ; <-ticker.C {
+		if config.StartIndex == nil || config.EndIndex == nil {
+			logInfo, err := rekor.GetLogInfo(context.Background(), rekorClient)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "error getting log info: %v", err)
+				return
+			}
+
+			checkpoint, err := rekor.ReadLatestCheckpoint(logInfo)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "error reading checkpoint: %v", err)
+				return
+			}
+
+			var prevCheckpoint *util.SignedCheckpoint
+			prevCheckpoint, err = file.ReadLatestCheckpoint(config.LogInfoFile)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "reading checkpoint log: %v", err)
+				return
+			}
+
+			checkpointStartIndex, checkpointEndIndex := rekor.GetCheckpointIndices(logInfo, prevCheckpoint, checkpoint)
+			if config.StartIndex == nil {
+				config.StartIndex = &checkpointStartIndex
+			}
+			if config.EndIndex == nil {
+				config.EndIndex = &checkpointEndIndex
+			}
+		}
+
+		if *config.StartIndex >= *config.EndIndex {
+			fmt.Fprintf(os.Stderr, "start index %d must be strictly less than end index %d", *config.StartIndex, *config.EndIndex)
+		}
+
+		// TODO: This should subsequently read from the identity metadata file to fetch the latest index.
+		err := rekor.IdentitySearch(*config.StartIndex, *config.EndIndex, rekorClient, config.MonitoredValues, config.OutputIdentitiesFile, config.IdentityMetadataFile)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to successfully complete identity search: %v", err)
+			return
+		}
+
+		if *once || config.EndIndex != nil {
+			return
+		}
+
+		config.StartIndex = config.EndIndex
```

Comment:
You'll need a variable initialized to `config.Start/EndIndex` at the beginning of the loop. Otherwise, `config.EndIndex` will get set in the conditional and then this will exit early. Does that seem correct?

Reply:
thanks for catching that, should be updated now